src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -208,7 +208,7 @@ public void ICollection_NonGeneric_CopyTo_TwoDimensionArray_ThrowsException(int
             }
         }
 
-        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
+        [ConditionalTheory("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
```

Comment:
Regarding the typeof() we would need to add an override for the attribute's constructor and handle the specific case of where the parameter is a type or a string. That way we could have the string one for local properties within the test assembly. 

A thing we should be careful of is that the attributes can have multiple conditions, so we should consider that when adding the typeof override. To support multiple conditions over that scenario.